### ISSUE TITLE
some numpy linalg matrix norms do not like negative axes

### ISSUE THREAD

**Issue by argriffing** [numpy/numpy#5725]
```python
>>> import numpy as np
>>> np.__version__
'1.10.0.dev0+a4cd474'
>>> I = np.eye(4)
>>> np.linalg.norm(I, axis=(0, 1))
2.0
>>> np.linalg.norm(I, axis=(-2, -1))
2.0
>>> np.linalg.norm(I, 'fro', axis=(-2, -1))
2.0
>>> np.linalg.norm(I, 1, axis=(0, 1))
1.0
>>> np.linalg.norm(I, 1, axis=(1, 0))
1.0
>>> np.linalg.norm(I, 1, axis=(-2, -1))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "[...]/numpy/linalg/linalg.py", line 2161, in norm
    ret = add.reduce(abs(x), axis=row_axis).max(axis=col_axis)
  File "[...]/numpy/core/_methods.py", line 26, in _amax
    return umr_maximum(a, axis, None, out, keepdims)
ValueError: 'axis' entry is out of bounds
```

**Comment by jaimefrio** [comment]
I think those would all be fixed if the `row_axis` and `col_axis` where made positive right after the check [here](https://github.com/numpy/numpy/blob/master/numpy/linalg/linalg.py#L2149). I don't fully grasp the logic, but some of those `if col_axis > row_axis: col_axis -= 1` further down will only work with positive axes.

**Comment by argriffing** [comment]
@jaimefrio That sounds plausible.  I'll leave this alone for a while, because it seems like a good opportunity for first-time contributors.

**Comment by jaimefrio** [comment]
Yes, I fully agree.

**Comment by gmcastil** [comment]
I'll take a look at this.

**Comment by jaimefrio** [comment]
@RittaNarita is already on it, see #5731.

**Comment by jaimefrio** [comment]
Closed by #5731.
